decryptor/mysql/response_proxy.go

```diff
@@ -386,30 +414,33 @@ func (handler *Handler) processTextDataRow(ctx context.Context, rowData []byte,
 	handler.logger.Debugln("Process data rows in text protocol")
 	for i := range fields {
 		fieldLogger = handler.logger.WithField("field_index", i)
-		value, _, n, err = LengthEncodedString(rowData[pos:])
+		value, n, err = LengthEncodedString(rowData[pos:])
 		if err != nil {
 			return nil, err
 		}
+		value, err = handler.onColumnDecryption(ctx, i, value)
```

Comment:
we try to decrypt here and below in `if handler.isFieldToDecrypt` branch? Maybe better to remove `if` check, and leave base logic with `onColumnDecryption`. And implement required interface by `handler` which will subscribe on `OnColumnDecryption` and do logic with `if handler.isFieldToDecrypt(fields[i])`?

We can encapsulate 
```
if handler.isFieldToDecrypt(fields[i]) {
			decryptedValue, err := handler.decryptor.DecryptBlock(value)
			if err == nil && decryptedValue != nil && len(decryptedValue) != len(value) {
				base.AcrastructDecryptionCounter.WithLabelValues(base.DecryptionTypeSuccess).Inc()
				fieldLogger.Debugln("Update with decrypted value")
				output = append(output, PutLengthEncodedString(decryptedValue)...)
				value = decryptedValue
			} else {
				if err != errPlainData {
					span.AddAttributes(trace.BoolAttribute("failed_decryption", true))
					base.AcrastructDecryptionCounter.WithLabelValues(base.DecryptionTypeFail).Inc()
				}
				fieldLogger.Debugln("Leave value as is")
				output = append(output, rowData[pos:pos+n]...)
			}
			pos += n
			continue
		} else {
			fieldLogger.Debugln("Field is not binary")
		}
```
to some `OnColumnEncryption` handler, subscribe him and leave here only `value, err = handler.onColumnDecryption(ctx, i, value)` to leave only one decryption

Reply:
> Maybe better to remove if check, and leave base logic with onColumnDecryption. 

I thought about it but left as is for now. As I understand, that's how PostgreSQL code handles decryption and it would be nice to unify that. I'll look into the diff, if it's reasonable I'll include a commit into this PR. It this requires more that just some code removal and fixups then I guess that should be a separate PR.

Reply:
Indeed, it is possible. I have pushed a commit which replaces hardwired decryption code in MySQL with using the OnColumn notifications. The common decryptor will now take care of that: check if a field can contain AcraStructs, try to decrypt them, update success/failure counters, etc. This also works for Acra EE.

Reply:
looks much better, less code) missed that already exists base code which do decryption and we will need create on more subscriber with checks over mysql binary types according to supported types by `isFieldToDecrypt`.

Reply:
> we will need create on more subscriber

Uh... Do we? The `OnColumn` is already executed for both text fields and binary fields. For text fields it is executed regardless of the field type: tokenization may be interested in integers, not just byte arrays, and decryption skips most of non-bytes fields because they are too small. For binary fields there are separate core paths, but again, decryption will be applied to numbers and will be a no-op. It's probably not ideal, but I guess it shouldn't impact performance too much.

Reply:
looked more thoughtful, questions disappeared and have not more questions. found where and how configured decryption for mysql in acra-ce and acra-ee